diagnostics/stratonovich_diagonal.py

```diff
@@ -0,0 +1,132 @@
+# Copyright 2020 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import os
+
+import argparse
+import matplotlib.pyplot as plt
+import numpy as np
+import torch
+import tqdm
+from scipy import stats
+
+from tests.problems import Ex2
+from torchsde import sdeint, BrownianInterval
+from torchsde.settings import LEVY_AREA_APPROXIMATIONS
+from .utils import to_numpy, makedirs_if_not_found, compute_mse
+
+
+def inspect_sample():
+    batch_size, d = 32, 1
+    steps = 100
+
+    ts = torch.linspace(0., 5., steps=steps).to(device)
```

Comment:
I know the current codebase isn't yet consistent on this point, but can this be changed to `torch.linspace(..., device=device)`?

Reply:
(and similarly elsewhere in this file where appropriate)

Reply:
Sure! Done
Left `.to(device)` only for `problems` instances. Can it stay that way?